common/oatbox/cache/GcpTokenCacheItemPool.php

```diff
@@ -0,0 +1,251 @@
+    {
+        $this->log(__METHOD__);
+
+        return $this->getCache()->clear();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function deleteItem($key)
+    {
+        $key = $this->getCacheKey();
+
+        $this->log(__METHOD__, [$key]);
+
+        return $this->getCache()->deleteItem($key);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function deleteItems(array $keys)
+    {
+        $keys = [$this->getCacheKey()];
+
+        $this->log(__METHOD__, $keys);
+
+        return $this->getCache()->deleteItems($keys);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function save(CacheItemInterface $item)
+    {
+        $this->log(__METHOD__);
+
+        if ($this->isWritingDisabled()) {
+            return true;
+        }
+
+        $this->serializeToPersist($item);
+
+        return $this->getCache()->save($item);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function saveDeferred(CacheItemInterface $item)
+    {
+        $this->log(__METHOD__);
+
+        if ($this->isWritingDisabled()) {
+            return true;
+        }
+
+        $this->serializeToPersist($item);
+
+        return $this->getCache()->saveDeferred($item);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function commit()
+    {
+        $this->log(__METHOD__);
+
+        if ($this->isWritingDisabled()) {
+            return true;
+        }
+
+        return $this->getCache()->commit();
+    }
+
+    private function log(string $method, array $keys = []): void
+    {
+        if ($this->isDebug()) {
+            $this->getLogger()->debug(
+                sprintf(
+                    '[%s] Called method %s with key(s) %s',
+                    __CLASS__,
+                    $method,
+                    implode(',', $keys)
+                )
+            );
+        }
+    }
+
+    private function isDebug(): bool
+    {
+        if ($this->isDebug === null) {
+            $this->isDebug = (bool)$this->getOption(self::OPTION_ENABLE_DEBUG, false);
+        }
+
+        return $this->isDebug;
+    }
+
+    private function isWritingDisabled(): bool
+    {
+        if ($this->isWritingDisabled === null) {
+            $this->isWritingDisabled = (bool)$this->getOption(self::OPTION_DISABLE_WRITE, false);
+        }
+
+        return $this->isWritingDisabled;
+    }
+
+    private function getCache(): CacheItemPoolInterface
+    {
+        $this->cache = $this->createCache();
+
+        return $this->cache;
+    }
+
+    private function createCache(): CacheItemPoolInterface
+    {
+        return $this->getCacheFactory()->create(
+            [
+                CacheItemPoolFactory::CONFIG_PERSISTENCE => $this->getOption(self::OPTION_PERSISTENCE, 'redis'),
+            ]
+        );
+    }
+
+    private function unSerializeIfNecessary(CacheItemInterface $item): void
+    {
+        if (empty($item->get())) {
+            return;
+        }
+
+        $unSerialized = @unserialize($item->get());
```

Comment:
I have some fear about this method. In the case of some snippets are caching serialized data, it will be unserialized here and a second time in the usage of this cache. WDYT?

Reply:
This cache is used only for GCP token, not other code. I would prefer json_encoded, but It is not possible right now. Do you have any suggestion that would work with this?

Reply:
If its only used by GCP, the risk is low 👍